router/router_system_test.go

```
@@ -1,17 +1,209 @@
+			System: system.System{
+				Architecture:  "amd64",
+				CPUThreads:    4,
+				KernelVersion: "6.8.0",
+				OSType:        "linux",
+			},
+		}, nil
+	}
+
+	recorder := httptest.NewRecorder()
+	c, _ := gin.CreateTestContext(recorder)
+	c.Request = httptest.NewRequest(http.MethodGet, "/api/system", nil)
```

golangci-lint (2.13.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)